pfioh/store_options.py

```diff
@@ -0,0 +1,348 @@
+            # except:
+            self.qprint('<transmission>', comms = 'tx')
+            d_ret['transmit']               = {}
+            d_ret['transmit']['msg']        = 'transmitting'
+            d_ret['transmit']['timestamp']  = '%s' % datetime.datetime.now()
+            d_ret['transmit']['filesize']   = '%s' % os.stat(str_fileToProcess).st_size
+            d_ret['status']                 = True
+            d_ret['msg']                    = d_ret['transmit']['msg']
+            self.wfile.write(fh.read())
+
+        #Cleanup by deleting temporary files
+        if b_cleanup:
+            if b_zip:
+                self.qprint("Removing '%s'..." % (str_zipFile), comms = 'status')
+                if os.path.isfile(str_zipFile):     os.remove(str_zipFile)
+            if str_encoding == 'base64':
+                self.qprint("Removing '%s'..." % (str_base64File), comms = 'status')
+                if os.path.isfile(str_base64File):  os.remove(str_base64File)
+
+        return d_ret
+
+
+class SwiftStore(StoreHandler):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.qprint('SwiftStore initialized')
+        
+
+    def _getScopedSession(self, osAuthUrl, username, password, osProjectDomain, osProjectName):
+        """
+        Uses keystone authentication to create and return a scoped session
+        """
+
+        passwordAuth = v3.Password(auth_url= osAuthUrl,
+                            user_domain_name= 'default',
+                            username= username, password= password,
+                            project_domain_name= osProjectDomain,
+                            project_name= osProjectName,
+                            unscoped= False)
+
+        scopedSession = session.Session(auth= passwordAuth)
+        return scopedSession
+
+
+    def _initiateSwiftConnection(self, **kwargs):
+        """
+        Initiates a Swift connection and returns a Swift Connection Object
+        """
+
+        for k,v in kwargs:
+            if k== 'configPath': configPath= v
+
+        #Paramters will be initialized by configmap
+        credentialsDict= self._getSwiftCredentials(configPath)
+        osAuthUrl= credentialsDict['osAuthUrl']
+        username= credentialsDict['username']
+        password= credentialsDict['password']
+        osProjectDomain= credentialsDict['osProjectDomain']
+        osProjectName= credentialsDict['osProjectName']
+
+        scopedSession = self._getScopedSession(osAuthUrl, username, password, osProjectDomain, osProjectName)
+        swiftConnection = swift_client.Connection(session=scopedSession)
+        return swiftConnection
+
+
+    def _getSwiftCredentials(self, configPath):
+        """
+        Initializes the swift storage parameters by accessing configmap
+        """
+
+        with open(configPath,'r') as f:
+            line= f.read()
+
+        credentialsDict= ast.literal_eval(line)

```

Comment:
I am not exactly sure about the format. I created the file like a simple dictionary. Should it be like .kube/config?